4.3: Properly update TinyMCE, CMFEditions, Sunburst

mauritsvanrees opened the pull request:
This fixes issue https://github.com/plone/Products.CMFPlone/issues/812

It fixes existing upgrade steps.

I think we should also explicitly run those upgrade steps again in Plone 4.3.7, to make sure these three packages are really updated for everyone. I added commit ba0f141f84087a30cc7dee1675c951a3b0bbf611 for that, but reverted it as it needs an update to metadata.xml in CMFPlone. Now it is available for easy cherry-picking.

mauritsvanrees commented:
http://jenkins.plone.org/job/pull-request-4.3/70/

mauritsvanrees commented:
http://jenkins.plone.org/job/pull-request-5.0/212/

mauritsvanrees commented:
The 4.3 Jenkins job is fine.
The 5.0 one gives errors, but they seem unrelated, things like `AttributeError: 'DB' object has no attribute 'storage'` in plone.testing.
Trying again:
http://jenkins.plone.org/job/pull-request-5.0/213/

mauritsvanrees commented:
The second try at Plone 5.0 fails too on Jenkins: problems when running the tests for plone.testing and plone.app.testing.  Locally all is fine, at least when running those tests separately. I can think of nothing in my code that causes this. Maybe something strange on Jenkins, although that is always easy to say...
Anyway, seems it is time for someone to review this pull request.

gforcada commented:
@mauritsvanrees if you look at the full log http://jenkins.plone.org/job/pull-request-5.0/213/consoleFull there is at least one traceback that gives a ``AttributeError: Products.TinyMCE`` and a second one that ends with

```
ConfigurationExecutionError: <type 'exceptions.KeyError'>: u'Duplicate profile ID: Products.DCWorkflow:revision2'
  in:
  File "/home/jenkins/.buildout/eggs/Products.DCWorkflow-2.2.4-py2.7.egg/Products/DCWorkflow/configure.zcml", line 21.2-27.8
    <genericsetup:registerProfile
        name="revision2"
        title="CMF Default Workflow [Revision 2]"
        description="Adds revision 2 of default workflow."
        provides="Products.GenericSetup.interfaces.EXTENSION"
        for="Products.CMFCore.interfaces.ISiteRoot"
        />
```

Maybe that's a bit more insightful?

mauritsvanrees commented:
@gforcada Thanks, that helps.  And I should have seen that locally when testing with Plone 5, but I was focused on Plone 4...

Strange: that test failure does not actually show up in the page with the failed tests: http://jenkins.plone.org/job/pull-request-5.0/213/testReport/

mauritsvanrees commented:
I have fixed the TinyMCE problem.
I think I will hope that the DCWorkflow one then magically goes away...
I will fire up a new Jenkins job once Jenkins is awake -- some out of space errors currently.

mauritsvanrees commented:
Running Jenkins at http://jenkins.plone.org/job/pull-request-5.0/217/

mauritsvanrees commented:
Jenkins for the Plone 5 pull request is happy now.
Redoing the 4.3 on Jenkins to be sure: http://jenkins.plone.org/job/pull-request-4.3/71/

mauritsvanrees commented:
Okay, Jenkins is happy on Plone 4.3 and 5.0.

So if someone can review this, it would be good.

mauritsvanrees commented:
Review and comments on the general direction are welcome.
But I will meanwhile add some more fixes, though they could be done separately just on master.
Also, I will see about using the portal_setup.upgradeProfile method that I introduced in GenericSetup 1.7.7.

mauritsvanrees commented:
Jenkins is running new jobs:
http://jenkins.plone.org/job/pull-request-4.3/73/
http://jenkins.plone.org/job/pull-request-5.0/222/

mauritsvanrees commented:
Both Jenkins tests pass.
Please review.

jensens commented:
@mauritsvanrees looks fine and as i get it it fixes an upgrade bug i know well. 
please rebase and merge yourself afterwards

mauritsvanrees commented:
I thought this was already merged a while ago, but apparently I imagined this. Thanks for looking into it.

I have rebased and squashed it into one commit.

Jenkins job for 4.3: http://jenkins.plone.org/job/pull-request-4.3/98/
Jenkins job for 5.0: http://jenkins.plone.org/job/pull-request-5.0/357/

mauritsvanrees commented:
Green on both Jenkins jobs.
